sinks/gcm/sink.go

```diff
@@ -31,24 +37,233 @@ func NewSink() (*GcmSink, error) {
 		return nil, err
 	}
 
+	// Get supported metrics.
+	supportedMetrics := allMetrics[0:]
+	for i := range supportedMetrics {
+		supportedMetrics[i].Labels = allLabels
+	}
+
+	// Create the metrics.
+	descriptors := make([]MetricDescriptor, 0, len(supportedMetrics))
+	for _, supported := range supportedMetrics {
+		descriptors = append(descriptors, supported.MetricDescriptor)
+	}
+	err = driver.AddMetrics(descriptors)
+	if err != nil {
+		return nil, err
+	}
+
 	return &GcmSink{
-		driver: driver,
+		driver:           driver,
+		supportedMetrics: supportedMetrics,
 	}, nil
 }
 
-func (self *GcmSink) StoreData(data interface{}) error {
-	data, ok := data.(sources.ContainerData)
+func (self *GcmSink) StoreData(input interface{}) error {
+	data, ok := input.(sources.ContainerData)
 	if !ok {
 		return fmt.Errorf("requesting unrecognized type to be stored in GCM")
 	}
 
-	// TODO(vmarmol): Implement.
+	// Get metrics from the raw data.
+	metrics := make([]Metric, 0, len(data.Pods)+len(data.Containers)+len(data.Machine))
+	for _, pod := range data.Pods {
+		metrics = append(metrics, self.podToMetrics(&pod)...)
+	}
+	metrics = append(metrics, self.rawContainersToMetrics(data.Containers)...)
+	metrics = append(metrics, self.rawContainersToMetrics(data.Machine)...)
+
+	// Push the metrics step size at a time. Try to push all the metrics even if there are errors.
+	var lastErr error
+	step := self.driver.MaxNumPushMetrics()
+	for i := 0; i < len(metrics); i += step {
+		endIndex := i + step
+		if endIndex > len(metrics) {
+			endIndex = len(metrics)
+		}
+
+		err := self.driver.PushMetrics(metrics[i:endIndex])
+		if err != nil {
+			lastErr = err
+		}
+	}
+
+	return lastErr
+}
+
+// Concatenates a map of labels into a comma-separated key=value pairs.
+func labelsToString(labels map[string]string) string {
+	var buffer bytes.Buffer
+	first := true
+	for key, value := range labels {
+		if !first {
+			buffer.WriteString(",")
+		}
+		buffer.WriteString(key)
+		buffer.WriteString("=")
+		buffer.WriteString(value)
+		first = false
```

Comment:
Do you mean to put it inside the if? It won't trigger the first time. Moved it closer to the if in case that is what you meant :)